_posts/2016-08-10-dynamic-loading.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Dynamic Code Loading on a MCU
+authors:
+  - brghena
+  - alevy
+---
+
+One key feature of Tock is the ability to load and run multiple applications
+simultaneously. In a modern computer, the OS uses the Memory Management Unit
+(MMU) to provide a virtual address space for each process. The code for each
+process is written assuming that it is the only code in the world and that it
+can place memory at any address it pleases. The MMU handles translating these
+virtual addresses into a physical address in real memory, which is shared
+between all processes. Unfortunately, in the world of embedded systems MMUs are
+not available. Processors like the ARM Cortex-M series omit them because they
+aren't necessary for traditional single-process embedded devices.
```

Comment:
That's not why they omit them. They're omitted because they are power and area-hungry hardware elements that are complex and costly to design. I think the fact that an MPU was created at all is evidence that there is desire for isolation traditionally provided by an MMU, and if it were as simple as just copying the HW element over from another design that would have been done long ago.